camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/DslComponent.java

```diff
@@ -84,51 +89,72 @@ public DslComponent(DslComponent scopeComponent, Scope scope, String componentId
     }
 
     // ---------------------------
-    
+
     @Override
-    public Task<Entity> newTask() {
-        return TaskBuilder.<Entity>builder().displayName(toString()).tag(BrooklynTaskTags.TRANSIENT_TASK_TAG)
-            .body(new EntityInScopeFinder(scopeComponent, scope, componentId)).build();
+    public Task<O> newTask() {
+        return TaskBuilder.<O>builder()
+                .displayName(toString())
+                .tag(BrooklynTaskTags.TRANSIENT_TASK_TAG)
+                .body(new ObjectInScopeFinder(scopeComponent, scope, componentId))
+                .build();
     }
-    
-    protected static class EntityInScopeFinder implements Callable<Entity> {
+
+    protected static class ObjectInScopeFinder<O> implements Callable<O> {
         protected final DslComponent scopeComponent;
         protected final Scope scope;
         protected final String componentId;
 
-        public EntityInScopeFinder(DslComponent scopeComponent, Scope scope, String componentId) {
+        public ObjectInScopeFinder(DslComponent scopeComponent, Scope scope, String componentId) {
             this.scopeComponent = scopeComponent;
             this.scope = scope;
             this.componentId = componentId;
         }
 
         protected EntityInternal getEntity() {
             if (scopeComponent!=null) {
-                return (EntityInternal)scopeComponent.get();
+                return (EntityInternal) scopeComponent.get();
             } else {
                 return entity();
             }
         }
-        
+
+        protected LocationInternal getLocation() {
+            if (scopeComponent!=null) {
+                return (LocationInternal) scopeComponent.get();
+            } else {
+                throw new IllegalStateException("Scope component must be set");
+            }
+        }
+
         @Override
-        public Entity call() throws Exception {
-            Iterable<Entity> entitiesToSearch = null;
+        public O call() throws Exception {
             EntityInternal entity = getEntity();
```

Comment:
I don't follow what the expectation of this class is now. Do we expect the `scopeComponent` to always be an entity? Or sometimes a location? If it calls `call()` then it will fail for anything but an entity, I presume.